src/mc-range-edge-generation-private.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MC_RANGE_EDGE_GENERATION_PRIVATE_H
+#define MC_RANGE_EDGE_GENERATION_PRIVATE_H
+
+#include <stddef.h> // size_t
+#include <stdint.h>
+#include "mc-optional-private.h"
+#include "mongocrypt-status-private.h"
+
+// mc_edges_t represents a list of edges.
+typedef struct _mc_edges_t mc_edges_t;
```

Comment:
Is it necessary that `mc_edges` have its definition hidden and implemented out-of-line?

Reply:
Not necessary. I generally prefer limiting API in private headers unless it is an inconvenience.